src/rpc/blockchain.cpp

```diff
@@ -2284,6 +2284,108 @@ static RPCHelpMan scantxoutset()
+                            },
+                        },
+                        "[scanobjects,...]"},
+                    {"startheight", RPCArg::Type::NUM, /*default*/ "0", "height to start to filter from"},
+                    {"filtertype", RPCArg::Type::STR, /*default*/ "basic", "The type name of the filter"}
+                },
+                RPCResult{
+                    RPCResult::Type::ARR, "", "",
+                    {
+                        {RPCResult::Type::STR_HEX, "", "The blockhash"},
+                    }},
+                RPCExamples{
+                    HelpExampleCli("scanblockfilters", "[\"addr(bcrt1q4u4nsgk6ug0sqz7r3rj9tykjxrsl0yy4d0wwte)\"] 300000") +
+                    HelpExampleRpc("scanblockfilters", "[\"addr(bcrt1q4u4nsgk6ug0sqz7r3rj9tykjxrsl0yy4d0wwte)\"] 300000")
+                },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    std::string filtertype_name = "basic";
+    if (!request.params[2].isNull()) {
+        filtertype_name = request.params[1].get_str();
+    }
+
+    BlockFilterType filtertype;
+    if (!BlockFilterTypeByName(filtertype_name, filtertype)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Unknown filtertype");
+    }
+
+    BlockFilterIndex* index = GetBlockFilterIndex(filtertype);
+    if (!index) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Index is not enabled for filtertype " + filtertype_name);
+    }
+
+    // set the stop-height
+    const CBlockIndex* block = nullptr;
+    {
+        LOCK(cs_main);
+        block = ::ChainActive().Tip();
+        if (!request.params[1].isNull()) {
+            block = ::ChainActive()[request.params[1].get_int()];
+            if (!block) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Invalid stopheight");
+            }
+        }
+    }
+    CHECK_NONFATAL(block);
+
+    // loop through the scan objects, add scripts to the element_set
+    GCSFilter::ElementSet needle_set;
+    for (const UniValue& scanobject : request.params[0].get_array().getValues()) {
+        FlatSigningProvider provider;
+        std::vector<CScript> scripts = EvalDescriptorStringOrObject(scanobject, provider);
+        for (const CScript& script : scripts) {
+            needle_set.emplace(script.begin(), script.end());
+        }
+    }
+    NodeContext& node = EnsureNodeContext(request.context);
+    UniValue ret(UniValue::VARR);
+    const int amount_per_chunk = 10000;
+    const CBlockIndex* start_index = block;
+    std::vector<BlockFilter> filters;
+    while (block) {
+        node.rpc_interruption_point(); // allow a clean shutdown
+        const CBlockIndex* next = nullptr;
+        {
+            LOCK(cs_main);
+            next = ChainActive().Next(block);
+        }
+        if (start_index->nHeight+amount_per_chunk == block->nHeight || next == nullptr) {
+            LogPrint(BCLog::RPC, "Fetching blockfilters from height %d to height %d.\n", start_index->nHeight, block->nHeight);
```

Comment:
I think this is worth logging in general rather than RPC only, just like `rescanwallet`.